src/extension.ts

```diff
@@ -258,34 +244,38 @@ export async function activate(extensionContext: ExtensionContext): Promise<unkn
         }
     }
 
-    OpenShiftExplorer.getInstance().onDidChangeContextEmitter.event(async (context) => {
-        const kind = await getNamespaceKind();
-        void Oc.Instance.getActiveProject().then((namespace) =>
-            updateContextStatusBarItem(activeNamespaceStatusBarItem, 'project-node', namespace, `Current ${kind}: ${namespace}`,
-                !isNamespaceInfoStatusBarDisabled()));
+    const updateContextAndProjectStatusBarItems = async function(context?: string): Promise<void> {
+        return getNamespaceKind().then(async (kind) => {
```

Comment:
this function is async. You can await `getNamespaceKind` instead of using `.then(...`

Reply:
ACK